python/ray/_private/runtime_env/container.py

```diff
@@ -33,10 +34,13 @@ async def setup(
             "--network=host",
             "--pid=host",
             "--ipc=host",
-            "--env-host",
+            "--user=root",
```

Comment:
should we make this configurable? or root is necessary?

Reply:
There are some files in the mounted folder `/tmp/ray` that the `default_worker.py` process needs to access inside the container, so it complains if I don't run the container as root because of permissions issues. Not sure how this worked originally without root user, this was originally written 2 years ago and there were no tests, so a lot of things in core probably changed during that time.

Happy to dig into it more if you have time for a pair session. For now, I added this because it doesn't work without it.

Reply:
We'd try to avoid elevating permissions to root

Which would mean we need to carefully manage permissions of things in the `/tmp/ray`